src/lib/generateDocument/generateDocument.test.ts

```diff
@@ -0,0 +1,228 @@
+// of this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+// Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+/* eslint-disable-next-line no-shadow */
+import { describe, expect, it } from "@jest/globals";
+import generateDocument from "./generateDocument";
+
+const DEFAULT_NAME = "My App";
+const DEFAULT_CLIENT_IDENTIFIER = "https://app.example/id";
+const DEFAULT_CLIENT_URI = "https://app.example/about";
+const DEFAULT_REDIRECT_URI = "https://app.example/callback";
+const DEFAULT_LOGO_URI = "https://app.example/logo";
+const DEFAULT_TOS_URI = "https://app.example/tos";
+const DEFAULT_POLICY_URI = "https://app.example/policy";
+const DEFAULT_CONTACT = "maintainer@app.example";
+
+describe("generateDocument creates correct Client Identifier Documents", () => {
+  it("creates full document", async () => {
+    const clientIdDocument = generateDocument({
+      clientId: DEFAULT_CLIENT_IDENTIFIER,
+      clientName: DEFAULT_NAME,
+      clientUri: DEFAULT_CLIENT_URI,
+      redirectUris: [DEFAULT_REDIRECT_URI],
+      useRefreshTokens: false,
+      logoUri: DEFAULT_LOGO_URI,
+      tosUri: DEFAULT_TOS_URI,
+      policyUri: DEFAULT_POLICY_URI,
+      contact: DEFAULT_CONTACT,
+      applicationType: "web",
+      requireAuthTime: true,
+      defaultMaxAge: 60 * 60,
+      compact: false,
+    });
+    const expected = `{
+  "@context": [
+    "https://www.w3.org/ns/solid/oidc-context.jsonld"
+  ],
+  "client_id": "https://app.example/id",
+  "client_name": "My App",
+  "client_uri": "https://app.example/about",
+  "redirect_uris": [
+    "https://app.example/callback"
+  ],
+  "grant_types": [
+    "authorization_code"
+  ],
+  "scope": "openid webid",
+  "token_endpoint_auth_method": "none",
+  "logo_uri": "https://app.example/logo",
+  "tos_uri": "https://app.example/tos",
+  "policy_uri": "https://app.example/policy",
+  "contacts": [
+    "maintainer@app.example"
+  ],
+  "application_type": "web",
+  "require_auth_time": true,
+  "default_max_age": 3600
+}`;
+    expect(clientIdDocument).toBe(expected);
```

Comment:
For this test to be more robust, I would parse the generated string and the expected object as JSON, ad do a field-by-field comparison. This way, if for some reason the order of the fields is modified in the generator for instance, the test remains valid. This remark applies to the other tests in this file too. 

In addition, you can add a test that no fields in the generated document is not present in the expected document.

Reply:
Regarding the "no field is not present", that's what I'd test in the `creates full document` test, right?

Reply:
The `creates full document` test verifies that there is no missing information in the generated client identifier, which is a good thing. What I'm suggesting is an additional test, which checks that the generated document only has the fields you expect, and nothing more.